server/aesdsocket.c

```diff
@@ -0,0 +1,355 @@
+  {
+    perror("server bind fd");
+    return -1;
+  }
+ #if DAEMON_CODE 
+  /* Daemon creation*/
+  if(set_daemon == 1)
+  {
+    syslog(LOG_DEBUG,"daemon\n");
+    pid = fork();
+    if(pid == -1)
+    {
+      perror("fork failed");
+      return -1;
+    }
+    else if(pid > 0)
+    {
+      //parent context
+      syslog(LOG_DEBUG,"CHILD PID = %d\n",pid);
+      exit(EXIT_SUCCESS);
+    }
+    
+    /* create new session and process grp*/
+    if(setsid()== -1)
+    {
+      perror("set sid failure");
+      return -1;
+    }
+    
+    //change cd to root
+    if(chdir("/") == -1)
+    {
+      perror("chdir");
+      return -1;
+    }
+    
+    //close all open files (in,out,error)
+    close(STDIN_FILENO);
+    close(STDOUT_FILENO);
+    close(STDERR_FILENO); 
+
+    open("/dev/null",O_RDWR);
+    dup(0);
+    dup(0);
+  }
+  #endif
+  
+  /* listen */
+  int server_listen_fd = 0;
+  
+  server_listen_fd = listen(server_socket_fd,BACK_LOG);
+  if(server_listen_fd == -1)
+  {
+    perror("server_listen_fd");
+    return -1;
+  }
+  
+  /* file open logic */
+  int ret_status = 0;
+  mode_t mode = S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH;
+  char* filename = FILE_PATH_TO_WRITE;
+  file_des = open(filename, (O_RDWR | O_CREAT), mode);
+  if(file_des == -1) //file_ptr
+  {
+    syslog(LOG_ERR,"file creation,opening failure\n");
+    perror("file creation");
+    exit(1);
+  }
+  
+  /* local declarations */
+  int current_data_pos = 0;
+  int read_status = 0;
+  int no_of_bytes_rcvd = 0;
+  
+  char* writer_file_buffer_ptr = NULL;
+  int write_buffer_size = BUFFER_CAPACITY;
+  writer_file_buffer_ptr = (char *)malloc(sizeof(char)*BUFFER_CAPACITY);
+  memset(writer_file_buffer_ptr,'\0',BUFFER_CAPACITY);
+  if(writer_file_buffer_ptr == NULL)
+  {
+    perror("writer_file_buffer");
+  }
+  
+  char* read_file_buffer_ptr = NULL;
+  bool run_status = true;
+  
+  while(run_status)
+  {    
+    /* Accept */
+    int client_accept_fd = 0;
+    socklen_t server_address_len = 0;
+    server_address_len = sizeof(server_address);
+    char s[INET6_ADDRSTRLEN];
+    memset(&s[0],'\0',sizeof(s));
+    
+    client_accept_fd = accept(server_socket_fd, (struct sockaddr *)&server_address,&server_address_len);
+    /* Accepted connection from client */
+    inet_ntop(AF_INET,get_in_addr((struct sockaddr *)&server_address),s,sizeof(s));
+    syslog(LOG_DEBUG, "Accepted connection from %s", s);
+    if(client_accept_fd == -1)
+    {
+      perror("server_accept_fd");
+      continue;
+    }
+    
+    /* sig status */
+    int sig_status = 0;
+    sig_status = sigprocmask(SIG_BLOCK, &x, NULL);
+    if(sig_status == -1)
+    {
+      perror("sig_status - 1");
+    }
+    
+    /* read */
+    int curr_location = 0;
+    while((no_of_bytes_rcvd = (recv(client_accept_fd, writer_file_buffer_ptr + curr_location, BUFFER_CAPACITY, 0))))
+    {
+      if(no_of_bytes_rcvd + curr_location >= write_buffer_size)
+      {
+        write_buffer_size *= MULTIPLIER_FACTOR; 
```

Comment:
@Rishab-Shah as we discussed, I think you could potentially simplify the logic here.